Source/Foundation/bsfUtility/Reflection/BsRTTIType.h

```diff
@@ -70,211 +70,211 @@ namespace bs
-  * variable. 
+  * Same as BS_RTTI_MEMBER_REFLPTR_ARRAY, but allows you to specify separate names for the field name and the member
+  * variable.
   */
-#define BS_RTTI_MEMBER_REFLPTR_ARRAY_NAMED(name, field, id)										\
-	META_Entry_##name;																			\
+#define BS_RTTI_MEMBER_REFLPTR_ARRAY_NAMED(name, field, id)																	\
+	META_Entry_##name;																					\
 																								\
-	std::common_type<decltype(OwnerType::field)>::type::value_type get##name(OwnerType* obj, UINT32 idx) { return obj->field[idx]; }				\
-	void set##name(OwnerType* obj, UINT32 idx, std::common_type<decltype(OwnerType::field)>::type::value_type val) { obj->field[idx] = val; }		\
-	UINT32 getSize##name(OwnerType* obj) { return (UINT32)obj->field.size(); }					\
-	void setSize##name(OwnerType* obj, UINT32 val) { obj->field.resize(val); }					\
+	std::common_type<decltype(OwnerType::field)>::type::value_type get##name(OwnerType* obj, UINT32 idx) { return obj->field[idx]; }							\
+	void set##name(OwnerType* obj, UINT32 idx, std::common_type<decltype(OwnerType::field)>::type::value_type val) { obj->field[idx] = val; }						\
+	UINT32 getSize##name(OwnerType* obj) { return (UINT32)obj->field.size(); }														\
+	void setSize##name(OwnerType* obj, UINT32 val) { obj->field.resize(val); }														\
 																								\
-	struct META_NextEntry_##name{};																\
-	void META_InitPrevEntry(META_NextEntry_##name typeId)										\
+	struct META_NextEntry_##name{};																				\
+	void META_InitPrevEntry(META_NextEntry_##name typeId)																	\
 	{																							\
-		addReflectablePtrArrayField(#name, id, &MyType::get##name, &MyType::getSize##name, &MyType::set##name, &MyType::setSize##name);			\
-		META_InitPrevEntry(META_Entry_##name());												\
+		addReflectablePtrArrayField(#name, id, &MyType::get##name, &MyType::getSize##name, &MyType::set##name, &MyType::setSize##name);							\
+		META_InitPrevEntry(META_Entry_##name());																	\
 	}																							\
 																								\
 	typedef META_NextEntry_##name
 
 /** Ends definitions for member fields with a RTTI type. Must follow BS_BEGIN_RTTI_MEMBERS. */
-#define BS_END_RTTI_MEMBERS																		\
-	META_LastEntry;																				\
+#define BS_END_RTTI_MEMBERS																					\
+	META_LastEntry;																						\
 																								\
-	struct META_InitAllMembers																	\
+	struct META_InitAllMembers																				\
 	{																							\
-		META_InitAllMembers(MyType* owner)														\
+		META_InitAllMembers(MyType* owner)																		\
 		{																						\
-			owner->META_InitPrevEntry(META_LastEntry());										\
+			owner->META_InitPrevEntry(META_LastEntry());																\
 		}																						\
 	};																							\
 																								\
-	META_InitAllMembers mInitMembers;
+	META_InitAllMembers mInitMembers{this};
```

Comment:
I think the only real change in this file is this line, and the comment at the top. The other changes seem to mess up alignment of the backslashes in macros, so I just reverted the old file and applied those changes manually, to avoid fixing all of them individually. Hopefully I didn't miss any other actual changes.

Reply:
Hrmm, well, actually I made all of those changes specifically because in my editor (I was using the "nano" command line editor), the back slashes were significantly mis-aligned.

I mean, each editor renders tabs differently, but I manually counted the tabs on a couple of lines, and came away thinking they weren't aligned based on the number of tab characters in the file. 

I don't care one way or another though, so whatever you want to do is fine with me.

Though, note that there were numerous lines below the macros that had extraneous spaces at the end of the line.

Reply:
If you can set up your tabs to be 4 spaces (or ideally, if nano can handle reading .editorconfig file) I think everything should be aligned (at least it is when I'm viewing it in VS).